paasta_tools/setup_marathon_job.py

```diff
@@ -471,15 +491,22 @@ def undrain_tasks(
 ) -> None:
     # If any tasks on the new app happen to be draining (e.g. someone reverts to an older version with
     # `paasta mark-for-deployment`), then we should undrain them.
-    for task in to_undrain:
+
+    async def undrain_task(task: MarathonTask) -> None:
         if task not in leave_draining:
             if task.state == 'TASK_UNREACHABLE':
-                continue
+                return
             try:
-                drain_method.stop_draining(task)
+                await drain_method.stop_draining(task)
             except Exception as e:
                 log_deploy_error("Ignoring exception during stop_draining of task %s: %s." % (task, e))
 
+    undrain_task_futures = []
+    for task in to_undrain:
+        undrain_task_futures.append(asyncio.ensure_future(undrain_task(task)))
+    if undrain_task_futures:
+        asyncio.get_event_loop().run_until_complete(asyncio.wait(undrain_task_futures))
```

Comment:
I've written some variation of these five lines several times now, and I feel like there must be a cleaner way to write this.

Reply:
Unfortunately when you are mixing async and non-async code you have to do this. You could use a map here

```
asyncio.get_event_loop().run_until_complete(asyncio.wait([
    asyncio.ensure_future(undrain_task(task)) for task in to_undrain
]))
```

If you don't care about the return value you could run this event loop in another thread